httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/CreateExchange.kt

```diff
@@ -27,23 +29,37 @@ import tbdex.sdk.protocol.models.Rfq
  * @param callback An optional callback function to be invoked after processing the RFQ.
  */
 @Suppress("TooGenericExceptionCaught", "SwallowedException")
-suspend fun submitRfq(
+suspend fun createExchange(
   call: ApplicationCall,
   offeringsApi: OfferingsApi,
   exchangesApi: ExchangesApi,
   callback: SubmitCallback?
 ) {
   val message: Rfq?
-
+  val replyTo: String?
   try {
-    message = Message.parse(call.receiveText()) as Rfq
+    val requestBody = call.receiveText()
+
+    val jsonNode = Json.jsonMapper.readTree(requestBody)
+    val rfqJsonString = jsonNode["rfq"].toString()
+
+    message = Message.parse(rfqJsonString) as Rfq
+    replyTo = jsonNode["replyTo"]?.takeIf { !it.isNull }?.asText()
+
   } catch (e: Exception) {
     val errorDetail = ErrorDetail(detail = "Parsing of TBDex message failed: ${e.message}")
     val errorResponse = ErrorResponse(listOf(errorDetail))
     call.respond(HttpStatusCode.BadRequest, errorResponse)
     return
   }
 
+  if (replyTo != null && !isValidUrl(replyTo)) {
+    val errorDetail = ErrorDetail(detail = "replyTo must be a valid URL")
+    val errorResponse = ErrorResponse(listOf(errorDetail))
+    call.respond(HttpStatusCode.BadRequest, errorResponse)
+    return
+  }
+
   try {
     exchangesApi.getExchange(message.metadata.exchangeId.toString())
     val errorDetail = ErrorDetail(detail = "RFQ already exists.")
```

Comment:
Tangential to the `replyTo` stuff, but can you update the `catch (_: Exception)` below to `catch (_: NoSuchElementException)` and add comment `// exchangesApi.getExchange throws if no existing exchange is found`. For a sec I was pretty confused why we were swallowing an error so egregiously.

Further musings: maybe this isn't the most graceful API for `ExchangeApi`. Personally I would return an empty list rather than `throw`. We can tackle that in a follow up.

Reply:
fixed - yea, this try catch is a bit confusing!

in this request handler to create exchange, we want to make sure that `exchangesApi.getExchange(exchangeId)` does *not* return an exchange. 

this try catch is actually hoping an `NoSuchElementException` is thrown when an exchange with the exchangeId is not found - which is what we want, because we want to create a *new* exchange

as you said, we should probably return an empty list if an exchange with the exchangeId is not found instead of throwing an exception.